### PR TITLE
Move getOrCreate logic into AuraApi

### PR DESCRIPTION
**Pull request by DarthMax**

### CODE REVIEW

File: graphdatascience/session/dedicated_sessions.py [hunk 1]
```diff
@@ -109,10 +106,8 @@ def _find_existing_session(self, session_name: str) -> Optional[SessionDetails]:
 
         # this will only occur for admins as we cannot resolve Aura-API client_id -> console_user_id we fail for now
         if len(matched_sessions) > 1:
-            users = [s.user_id for s in matched_sessions]
             raise RuntimeError(
-                "Admin users need to chose a unique session name for now."
-                f" Multiple sessions with the name `{session_name}` across multiple users exist. Sessions found for users: {users}"
+                f"The use has access to multiple session with the name `{session_name}`. Please specify the id of the session that should be deleted."
```
Comment (Mats-SX) [inline graphdatascience/session/dedicated_sessions.py:110]: should we include the ids in the error message? also: typo
```suggestion
                f"The user has access to multiple session with the name `{session_name}`: `{[s.id for s in matched_sessions]}`. Please specify the id of the session that should be deleted."
```

File: graphdatascience/session/dedicated_sessions.py [hunk 2]
```diff
@@ -164,38 +159,3 @@ def _construct_client(
...
-
-    def _check_cloud_location(self, existing_session: SessionDetails, cloud_location: Optional[CloudLocation]) -> None:
-        # cloud_location was only recently stored. if the existing session has none, we dont know if its the same or not
-        # (should be save to always check on 27.09.2024)
-        # if no cloud_location was provided the location is derived from the db -> no need to check
-        if existing_session.cloud_location and cloud_location and existing_session.cloud_location != cloud_location:
-            raise ValueError(
-                f"Session `{existing_session.name}` exists in a different cloud location. "
-                f"Current: `{existing_session.cloud_location}`, Requested: `{cloud_location}`."
-            )
```
Comment (Mats-SX) [inline graphdatascience/session/dedicated_sessions.py:201]: 🤩

**Review by Mats-SX — CHANGES_REQUESTED** [review]
a small typo in an error message. I also think we should changelog the warning feature. otherwise: beautiful!

File: graphdatascience/session/dedicated_sessions.py [hunk 3]
```diff
@@ -51,21 +51,18 @@ def get_or_create(
...
+        create_details = self._create_session(session_name, dbid, password, memory.value, ttl, cloud_location)
 
-            session_id = existing_session.id
-        else:
-            create_details = self._create_session(session_name, dbid, password, memory.value, ttl, cloud_location)
-            session_id = create_details.id
+        if create_details.expiry_date:
+            until_expiry: timedelta = create_details.expiry_date - datetime.now(timezone.utc)
+            if until_expiry < timedelta(days=1):
+                raise Warning(f"Session `{create_details.name}` is expiring in less than a day.")
```
[the remark this answers is not in the document]
Comment (DarthMax) [reply graphdatascience/session/dedicated_sessions.py:63]: That existed before. I just moved it

File: graphdatascience/session/dedicated_sessions.py [hunk 4]
```diff
@@ -51,21 +51,18 @@ def get_or_create(
...
-        existing_session = self._find_existing_session(session_name)
 
         # hashing the password to avoid storing the actual db password in Aura
         password = hashlib.sha256(db_connection.password.encode()).hexdigest()
 
-        if existing_session:
-            self._check_expiry_date(existing_session)
-            self._check_memory_configuration(existing_session, memory.value)
-            self._check_dbid(existing_session, dbid)
-            self._check_cloud_location(existing_session, cloud_location)
```
[the remark this answers is not in the document]
Comment (DarthMax) [reply graphdatascience/session/dedicated_sessions.py:63]: yes

File: graphdatascience/session/dedicated_sessions.py (hunk 1 above)
Comment (DarthMax) [reply graphdatascience/session/dedicated_sessions.py]: I don't think there is much use to listing the session ids here. The user will have to list sessions to find the correct one.

**Comment by Mats-SX** [comment]
Still missing a changelog

File: graphdatascience/session/gds_sessions.py [hunk 5]
```diff
@@ -82,6 +82,8 @@ def get_or_create(
         Retrieves an existing session with the given session name and database connection,
         or creates a new session if one does not exist.
 
+        If the session is close to expiration, a warning will be raised.
```
Comment (FlorentinD) [inline graphdatascience/session/gds_sessions.py:85]: opted to document the behavior in the ref docs

**Comment by Mats-SX** [comment]
Still missing a changelog entry

**Comment by FlorentinD** [comment]
> Still missing a changelog entry

the expiry warning was still not added in this PR but even in 1.11. Here it was only moved. 
Or do you think of a different changelog?

File: graphdatascience/session/dedicated_sessions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mats-SX) [reply graphdatascience/session/dedicated_sessions.py:63]: Oh okay my bad. all good then

**Review by Mats-SX — APPROVED** [review]

**Pull request merged**